### PR TITLE
Moved typedefs, created store item search.

### PR DESCRIPTION
**Pull request by HPT-I**
Så skulle det virke. Der foregik noget mærkeligt med den anden branch.

### CODE REVIEW

File: src/items.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    t.total_price =  t.count * t.product_price;
+    strcpy(t.name, input_item.name);
+
+    return t;
+
+}
+
+int items_compare_item_names(char* name_to_find, char* name_to_search) {
+
```
Comment (Bliztle) [inline src/items.c:116]: r/programminghorror. Take a look, i think i saw this.
simplifiy to just the premise in the if statement

**Comment by HPT-I** [comment]
Jeg kan apparently slette andres kommentarer, det er interessant. Nu er den if fixed

**Comment by HPT-I** [comment]
Do not ever merge this request.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -10,6 +10,10 @@ configure_file(settings.conf settings.conf COPYONLY)
 add_executable(settings settings.c)
 
 # Testing
+add_executable(test_items test_items.c items.c)
+target_link_libraries(test_items m)
```
Comment (Bliztle) [inline src/CMakeLists.txt:14]: Missing add_test

File: src/calc_test.c [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
+#include "assert.h"
 #include "calc.h"
 #include <math.h>
 
 int main()
 {
-    assert(calc_degrees_to_radians(45) == M_PI_4);
-    assert(calc_degrees_to_radians(90) == M_PI_2);
-    assert(calc_degrees_to_radians(180) == M_PI);
+
```
Comment (Bliztle) [inline src/calc_test.c:7]: What is the point of these changes? I assume there is one, just don't know it

File: src/items.c [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    // Initialising the fields that will be compared against in the  search loop, in order to prevent garbage memory
+    // creating unexpected results.
+    found_item_s best_item = {
+        .count = 0,
+        .price_per_unit = -1, // Set to -1 to differentiate first run.
+    };
+
+    int item_found = 0;
+
+    for (int ite = 0; ite < store->items_count; ite++) {
```
Comment (Bliztle) [inline src/items.c:22]: At a glance, naming this something other than simply 'i' suggests a special meaning, or possibly some variable set in another way, that doesn't just loop. We should probably stick with the normal conventions.

File: src/items.c [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+    double min_n = ceil((min_size / store_item_size));
+
+    if (store_item_size * min_n < max_size) return min_n;
+    return 0;
+}
+
+void items_filter_items(basket_item_s *basket, int basket_size, store_s *store) {
+
+    for (int ite = 0; ite < basket_size; ite++) {
```
Comment (Bliztle) [inline src/items.c:66]: See above

File: src/calc_test.c [hunk 6]
```diff
@@ -1,11 +1,18 @@
-#include <assert.h>
+#include "assert.h"
```
Comment (Bliztle) [inline src/calc_test.c:1]: change to <>

File: src/test_items.c [hunk 7]
```diff
@@ -0,0 +1,369 @@
+#include <assert.h>
+#include "api/parse.h"
+#include "stdio.h"
+#include "stdlib.h"
+#include "test_items.h"
+#include "items.h"
+#include "time.h"
+
+int main() {
+    test_items_filter_items();
```
Comment (Bliztle) [inline src/test_items.c:10]: add return 0;

**Review by Bliztle — APPROVED** [review]

**Pull request merged**